hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/EnhancementHelper.java

```diff
@@ -63,7 +65,8 @@ public static boolean includeInBaseFetchGroup(
 			return true;
 		}
 
-		return ! bootMapping.isLazy();
+		return collectionsInDefaultFetchGroupEnabled && ( value instanceof Collection )
+				|| ! bootMapping.isLazy();
```

Comment:
This is minor and feel free to ignore this comment. IMO we could enhance code readability by swapping the operand order as following:
```
return ! bootMapping.isLazy || ( value instanceof Collection ) && collectionsInDefaultFetchGroupEnabled;
```
because short-circuiting avoids unnecessary checking and implies the following preconditions:
* the new logic only applies when `bootMapping.isLazy()` returns true;
* the `collectionsInDefaultFetchGroupEnabled` checking only applies when value is of `Collection` type;

There is no performance implication, however, for `instanceof` has been super fast in modern JVM. E.g., https://stackoverflow.com/questions/103564/the-performance-impact-of-using-instanceof-in-java